File: src/main/java/com/enderzombi102/cmt/zone/ZoneActionData.java
package com.enderzombi102.cmt.zone;

import net.minecraft.util.Identifier;

public class ZoneActionData {

	// command
	public String enterCommand = null;
	public String leaveCommand = null;
	public String midCommand = null;
	// function
	public Identifier leaveFunction = null;
	public Identifier enterFunction = null;
	public Identifier midFunction = null;

	public ZoneActionData withEnterFunction(Identifier cmd) {
		this.enterFunction = cmd;
		return this;
	}

	public ZoneActionData withMidFunction(Identifier cmd) {
		this.midFunction = cmd;
		return this;
	}

	public ZoneActionData withLeaveFunction(Identifier cmd) {
		this.leaveFunction = cmd;
		return this;
	}

	public ZoneActionData withEnterCommand(String cmd) {
		this.enterCommand = cmd;
		return this;
	}

	public ZoneActionData withMidCommand(String cmd) {
		this.midCommand = cmd;
		return this;
	}

	public ZoneActionData withLeaveCommand(String cmd) {
		this.leaveCommand = cmd;
		return this;
	}

}

File: src/main/kotlin/com/enderzombi102/cmt/zone/ZoneActionData.kt
package com.enderzombi102.cmt.zone

import net.minecraft.util.Identifier

class ZoneActionData {
	// command
	var enterCommand: String? = null
	var leaveCommand: String? = null
	var midCommand: String? = null

	// function
	var leaveFunction: Identifier? = null
	var enterFunction: Identifier? = null
	var midFunction: Identifier? = null
	fun withEnterFunction(cmd: Identifier?): ZoneActionData {
		enterFunction = cmd
		return this
	}

	fun withMidFunction(cmd: Identifier?): ZoneActionData {
		midFunction = cmd
		return this
	}

	fun withLeaveFunction(cmd: Identifier?): ZoneActionData {
		leaveFunction = cmd
		return this
	}

	fun withEnterCommand(cmd: String?): ZoneActionData {
		enterCommand = cmd
		return this
	}

	fun withMidCommand(cmd: String?): ZoneActionData {
		midCommand = cmd
		return this
	}

	fun withLeaveCommand(cmd: String?): ZoneActionData {
		leaveCommand = cmd
		return this
	}
}
